integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/duplicateids/TestResource.java

```diff
@@ -0,0 +1,107 @@
+package io.quarkus.it.mongodb.panache.duplicateids;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+
+import org.junit.jupiter.api.Assertions;
+
+@Path("/testDuplicateId")
+public class TestResource {
+
+    @Inject
+    TestDuplicateIdRepository testDuplicateIdRepository;
+
+    @GET
+    @Path("imperative/entity")
+    public Response TestDuplicateIdEntity() {
+        List<TestDuplicateIdEntity> entities = getTestImperativeEntities();
+
+        // insert all
+        Assertions.assertEquals(0, TestDuplicateIdEntity.count());
+        TestDuplicateIdEntity.persist(entities);
+        Assertions.assertEquals(10, TestDuplicateIdEntity.count());
+
+        // Verify that the _id is set
+        verifyIdsSet(entities);
+
+        // varargs
+        TestDuplicateIdEntity entity11 = new TestDuplicateIdEntity("id11");
```

Comment:
Maybe i'm having a brain fart but I'm not seeing how this test works. The server should enforce uniqueness on the `_id` field so the deletes should only really delete one entity.